### DISCUSSION TITLE
Segmentation fault Error

### DISCUSSION THREAD

**Discussion by abhijit-inge** [google/or-tools#3525] [Build and configuration issues]
Hi,

So I was wondering if you could help me out with an issue i'm facing
1) I downloaded the package from source to generate the binaries 
 I used  the following commands to build the binaries.
```
cmake -S. -Bbuild -DBUILD_DEPS:BOOL=ON -DBUILD_PYTHON:BOOL=ON
cmake --build build
```

2) Under private/src/build/python folder contains all the Ortools python artifacts which i then manually copied over to python3.8/site-packages. I tried to run an import command 
```    from ortools.constraint_solver import pywrapcp``` which yeilded a segmentation fault error

3) Note i cannot use pip as it's restricted by my organization. I'm guessing i cannot just over the files and maybe i'm missing some steps. Could you help me out?

Additional question
If i download the wheel package as is for linux distribution, Since i cannot use pip install, can I somehow just use contents of that package and move it over to python3.8/site-packages? Will that be enough?

**Comment by lperron** [comment]
If you copy the files directly, most likely, dynamic loading of libraries do not work. You need to install the whl file.

**Reply by abhijit-inge** [reply]
will I be able to install the whl file generated by copying it over to a docker container that doesn't have access to the internet? If yes could you help me out on the right command to use? 
If internet connection is needed could you suggest a workaround?

**Reply by Mizux** [reply]
https://github.com/google/or-tools/blob/9270aa0d59910175be8df9d7247e10b7233b4686/cmake/python.cmake#L387-L388

Your docker image must also contains the following packages..
https://github.com/google/or-tools/blob/9270aa0d59910175be8df9d7247e10b7233b4686/ortools/python/setup.py.in#L45-L49

**Reply by abhijit-inge** [reply]
Thanks for the reply. Is there another way that you can think of? An approach that maybe  uses the binaries created?

**Reply by abhijit-inge** [reply]
Any insight on this ? I ask this because i believe wheel pip3 install <wheel file> pretty much just copy the wheel artifacts into site-packages/ortools/ unless i'm missing something
